server/src/main/java/org/cloudfoundry/identity/uaa/alias/EntityAliasHandler.java

```diff
@@ -208,4 +216,23 @@ public final Optional<T> retrieveAliasEntity(final T originalEntity) {
     protected abstract T updateEntity(final T entity, final String zoneId);
 
     protected abstract T createEntity(final T entity, final String zoneId) throws EntityAliasFailedException;
+
+    protected static <T extends EntityWithAlias> boolean isValidAliasPair(
+            @NonNull final T entity1,
+            @NonNull final T entity2
+    ) {
+        // check if both entities have an alias
+        final boolean entity1HasAlias = hasText(entity1.getAliasId()) && hasText(entity1.getAliasZid());
+        final boolean entity2HasAlias = hasText(entity2.getAliasId()) && hasText(entity2.getAliasZid());
+        if (!entity1HasAlias || !entity2HasAlias) {
+            return false;
```

Comment:
This part is not covered, please add a Unit or IT

Reply:
added test with commit 4321693